lib/rs/src/protocol/mod.rs

```diff
@@ -88,156 +113,157 @@ impl Type {
 
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
 pub enum MessageType {
-    MtCall = 0x01,
-    MtReply = 0x02,
-    MtException = 0x03,
+    Call = 0x01,
+    Reply = 0x02,
+    Exception = 0x03,
 }
 
 impl MessageType {
     pub fn from_num(num: u64) -> Option<MessageType> {
         match num {
-            0x01 => Some(MessageType::MtCall),
-            0x02 => Some(MessageType::MtReply),
-            0x03 => Some(MessageType::MtException),
+            0x01 => Some(MessageType::Call),
+            0x02 => Some(MessageType::Reply),
+            0x03 => Some(MessageType::Exception),
             _ => None,
         }
     }
 }
 
-pub trait Writeable {
-    fn write(&self, iprot: &Protocol, transport: &mut Transport) -> TResult<()>;
+pub trait ThriftTyped {
+    fn typ() -> Type;
+}
+
+pub trait Encode: ThriftTyped {
+    fn encode<P, T>(&self, &mut P, &mut T) -> Result<()>
+    where P: Protocol, T: Transport;
```

Comment:
I prefer `where` clauses on the same line if they're simple and they fit, although I guess that convention is still being agreed upon...